test_cases/stdlib/collections/check_counter.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from collections import Counter
+from typing_extensions import assert_type
+
+
+class Foo: ...
+
+
+# Test the constructor
+# assert_type(Counter(), Counter[Never, int])  # pyright derives "Unknown" instead of "Never"
+assert_type(Counter(foo=42.2), "Counter[str, float]")
+assert_type(Counter({42: "bar"}), "Counter[int, str]")
+assert_type(Counter([1, 2, 3]), "Counter[int, int]")
+
+int_c: Counter[str] = Counter()
+assert_type(int_c, "Counter[str, int]")
+assert_type(int_c["a"], int)
+int_c["a"] = 1
+int_c["a"] += 3
+int_c["a"] += 3.5  # type: ignore
+
+float_c = Counter(foo=42.2)
+assert_type(float_c, "Counter[str, float]")
+assert_type(float_c["a"], float)
+float_c["a"] = 1.0
+float_c["a"] += 3.0
+float_c["a"] += 42
+float_c["a"] += "42"  # type: ignore
+
+custom_c: Counter[str, Foo] = Counter()
+assert_type(custom_c, "Counter[str, Foo]")
+assert_type(custom_c["a"], Foo)
```

Comment:
At runtime this is actually an int though. I wonder if we need to make all these methods return `_C | int`.

Reply:
Doesn't this sort of problem apply to any Counter with a non-int value type, though? This seems like a fundamental problem with this PR.

Reply:
> At runtime this is actually an int though. I wonder if we need to make all these methods return _C | int.

I wonder whether type checkers support `__missing__`, in which case, this should happen automatically when we add it to the stubs. But returning `_C | int` makes some sense to me for getter methods.